tests/integration_tests/functional/test_cpu_features.py

```diff
@@ -186,25 +187,38 @@ def test_brand_string(test_microvm_with_api, network_config):
     "0xc0000083",  # MSR_CSTAR
     "0xc0000100",  # MSR_FS_BASE
     "0xc0000101",  # MSR_GS_BASE
+    # MSRs below are required only on T2A however,
+    # we are adding them to the common exception list to keep things simple
+    "0x834"     ,  # LVT Performance Monitor Interrupt Register
+    "0xc0010007",  # MSR_K7_PERFCTR3
+    "0xc001020b",  # Performance Event Counter MSR_F15H_PERF_CTR5
+    "0xc0011029",  # MSR_F10H_DECFG also referred to as MSR_AMD64_DE_CFG
 ]
 # fmt: on
 
+# CPU templates which support MSR test
+MSR_SUPPORTED_TEMPLATE = ["T2S", "T2A"]
 
-@pytest.mark.skipif(
-    PLATFORM != "x86_64", reason="CPU features are masked only on x86_64."
+
+@pytest.fixture(
+    name="msr_cpu_template",
+    params=set(SUPPORTED_CPU_TEMPLATES).intersection(MSR_SUPPORTED_TEMPLATE),
 )
+def msr_cpu_template_fxt(request):
+    """CPU template fixture for MSR read/write supported CPU templates"""
+    return request.param
+
+
 @pytest.mark.skipif(
-    cpuid_utils.get_cpu_vendor() != cpuid_utils.CpuVendor.INTEL,
-    reason="CPU templates are only available on Intel.",
+    PLATFORM != "x86_64", reason="CPU features are masked only on x86_64."
```

Comment:
do we still need this?

Reply:
not required, I think it was redundant in the original code as well since we were already checking vendor as INTEL